src/routers/companyRouter.ts

```diff
@@ -38,14 +38,10 @@ router.post(NUMBER_URL, async (req: Request, res: Response, next: NextFunction)
   const formValidator = new FormValidator();
   const companyNumberSanitizer = new CompanyNumberSanitizer();
   await new CompanySearchHandler(formValidator, companyNumberSanitizer).post(req, res).then((data) => {
-    // eslint-disable-next-line no-prototype-builtins
-    if (Object.prototype.hasOwnProperty.call(data, errorsConst)) {
-      res.render(`${routeViews}` + COMPANY_SEARCH_PAGE, data);
-    } else {
-      // eslint-disable-next-line no-unused-expressions
-      req.session?.setExtraData(COMPANY_PROFILE, data);
-      res.redirect(COMPANY_CONFIRM_URL);
-    }
+    req.session?.setExtraData(COMPANY_PROFILE, data);
+    res.redirect(COMPANY_CONFIRM_URL);
+  }).catch((data) => {
+    res.render(`${routeViews}` + COMPANY_SEARCH_PAGE, data);
```

Comment:
As part of `ECCT-546` we're going to be removing the `COMPANY_SEARCH_PAGE` from REA Web and redirecting to the `company lookup` service. Worth discussing how we're going to handle errors then.

Reply:
Don't think this will be a problem as rea won't have to deal with a the post operation on selecting the company. The common component will simply call a get on the confirm page with the company number as a parameter